openwisp_monitoring/device/base/models.py

```diff
@@ -258,13 +258,56 @@ def save_data(self, time=None):
             timeout=86400,  # 24 hours
         )
         if app_settings.WIFI_SESSIONS_ENABLED:
-            tasks.save_wifi_clients_and_sessions.delay(
-                device_data=self.data, device_pk=self.pk
-            )
+            self.save_wifi_clients_and_sessions()
 
     def json(self, *args, **kwargs):
         return json.dumps(self.data, *args, **kwargs)
 
+    def save_wifi_clients_and_sessions(self):
+        _WIFICLIENT_FIELDS = ['vendor', 'ht', 'vht', 'he', 'wmm', 'wds', 'wps']
+        WifiClient = load_model('device_monitoring', 'WifiClient')
+        WifiSession = load_model('device_monitoring', 'WifiSession')
+
+        active_sessions = []
+        interfaces = self.data.get('interfaces', [])
+        for interface in interfaces:
+            if interface.get('type') != 'wireless':
+                continue
+            interface_name = interface.get('name')
+            wireless = interface.get('wireless', {})
+            if not wireless or wireless['mode'] != 'access_point':
+                continue
+            ssid = wireless.get('ssid')
+            clients = wireless.get('clients', [])
+            for client in clients:
+                # Save WifiClient
+                client_obj, created = WifiClient.objects.get_or_create(
```

Comment:
This is also data which is queried often but shouldn't change as often, can you cache this too?
I think cache invalidation would be simple here, when save is called the cache can be invalidated.